daemon/src/wallpaper.rs

```diff
@@ -31,45 +28,57 @@ struct AnimationState {
 }
 
 #[derive(Debug)]
-pub struct AnimationToken {
+pub(super) struct AnimationToken {
     id: usize,
     transition_done: Arc<AtomicBool>,
 }
 
 impl AnimationToken {
-    pub fn is_transition_done(&self) -> bool {
+    pub(super) fn is_transition_done(&self) -> bool {
         self.transition_done.load(Ordering::Acquire)
     }
 
-    pub fn set_transition_done(&self, wallpaper: &Wallpaper) {
+    pub(super) fn set_transition_done(&self, wallpaper: &Wallpaper) {
         if wallpaper.has_animation_id(self) {
             self.transition_done.store(true, Ordering::Release);
         }
     }
 }
 
+struct FrameCallbackHandler {
+    cvar: Condvar,
+    time: Mutex<Option<u32>>,
```

Comment:
It's probably worth adding a comment that this `time`, that is received in the frame callback, doesn't really mean anything and in particular should not be used for any kind of frame timing. (For that purpose, you can use presentation-time + some logic to predict when a frame will be presented.)

Reply:
Ah, good point.